libbeat/cmd/instance/beat.go

```diff
@@ -95,8 +96,9 @@ type beatConfig struct {
 	Keystore      *common.Config `config:"keystore"`
 
 	// output/publishing related configurations
-	Pipeline   pipeline.Config `config:",inline"`
-	Monitoring *common.Config  `config:"xpack.monitoring"`
+	Pipeline      pipeline.Config `config:",inline"`
+	Monitoring    *common.Config  `config:"xpack.monitoring"`
+	MonitoringNew *common.Config  `config:"monitoring"`
```

Comment:
Let's call them `XPackMonitoring` and `Monitoring`.

Reply:
Since these are exported field names, are you sure it's okay to "break" the API like that?

Reply:
They are semi-exported :)

The beatConfig type is not exported, that is fields can not be accessed by other packages.
Yet the fields inside beatConfig are exported, so they can be correctly filled in via reflection when unpacking the config. The `config:"<name>"` determines the configuration key name. The `Monitoring` and `MonitoringNew` fields names are only for internal representation.